go/chat/boxer.go

```diff
@@ -1419,6 +1558,32 @@ func (b *Boxer) boxV2orV3(messagePlaintext chat1.MessagePlaintext, baseEncryptio
 		return nil, err
 	}
 
+	// Make pairwise MACs if there are any pairwise recipients supplied. Note
+	// that we still sign+encrypt with a signing key above. If we want
+	// repudiability, it's the caller's responsibility to provide a zero
+	// signing key or similar. Signing with a real key and also MAC'ing is
+	// redundant, but it will let us test the MAC code in prod in a backwards
+	// compatible way.
+	var pairwiseMACs map[keybase1.KID][]byte // nil by default
+	if len(pairwiseMACRecipients) > 0 {
```

Comment:
can this move out into a helper actually so we can put a `TraceTimed` on it for profling?

Reply:
I added one here, and also in `shouldPairwiseMAC` (which is the real hog, because it loads all the UPAKs).